src/components/LandDetailPage/ENSChip/ENSChip.tsx

```diff
@@ -2,40 +2,63 @@ import * as React from 'react'
 import { Loader } from 'decentraland-ui'
 import { locations } from 'routing/locations'
 import Chip from 'components/Chip'
-import { Props } from './ENSChip.types'
+import { Props, State } from './ENSChip.types'
 import './ENSChip.css'
+import UnsetENSContentModal from '../UnsetENSContentModal'
+
+export default class ENSChip extends React.PureComponent<Props, State> {
+  state: State = {
+    showConfirmationModal: false
+  }
 
-export default class ENSChip extends React.PureComponent<Props> {
   handleOnClick = () => {
     this.props.onNavigate(locations.activity())
   }
 
   handleOnIconClick = () => {
+    this.setState({ showConfirmationModal: true })
+  }
+
+  handleCancel = () => {
+    this.setState({ showConfirmationModal: false })
+  }
+
+  handleConfirm = () => {
     const { ens, onUnsetENSContent } = this.props
     onUnsetENSContent(ens)
+    this.setState({ showConfirmationModal: false })
   }
 
   render() {
-    const { ens, isLoading } = this.props
+    const { land, ens, isLoading } = this.props
 
     return (
-      <Chip
-        className="ENSChip"
-        text={
-          isLoading ? (
-            <>
-              {ens.subdomain}
-              <Loader active size="tiny" />
-            </>
-          ) : (
-            ens.subdomain
-          )
-        }
-        isActive={isLoading}
-        icon={isLoading ? '' : 'minus'}
-        onClick={isLoading ? this.handleOnClick : undefined}
-        onIconClick={this.handleOnIconClick}
-      />
+      <>
+        <Chip
+          className="ENSChip"
+          text={
+            isLoading ? (
+              <>
+                {ens.subdomain}
+                <Loader active size="tiny" />
+              </>
+            ) : (
+              ens.subdomain
+            )
+          }
+          isActive={isLoading}
+          icon={isLoading ? '' : 'minus'}
+          onClick={isLoading ? this.handleOnClick : undefined}
+          onIconClick={this.handleOnIconClick}
+        />
+        <UnsetENSContentModal
+          land={land}
+          ens={ens}
+          open={this.state.showConfirmationModal}
```

Comment:
Not very important but tend to destruct our props and states at the beginning of the file like we do with [props here](https://github.com/decentraland/builder/pull/1534/files#diff-46484747eefd33ff47ffc0413c52abb5fb52ffb19a704c972c7b5515467c7423R33), maybe we can move this to `const { showConfirmationModal } = this.state` ?

Reply:
Removed the state in the last change as the new modal has been connected to the store and is opened via openModal